### DISCUSSION TITLE
Md-editor-v3 has event drag to upload file?

### DISCUSSION THREAD

**Discussion by damcuongcak3** [imzbf/md-editor-v3#348] [Q&A]
Can i use event drag in <MdEditor>

**Comment by damcuongcak3** [comment]
One more question. Can I edit mdpreview directly. i don't want to show meditor. 🙇‍♂️

**Reply by imzbf** [reply]
Sorry, this mode has been abandoned because there is not enough cost(More about technology and manpower).

**Comment by imzbf** [comment]
> Can i use event drag in

I need to investigate this feature

**Comment by imzbf** [comment]
I add an event named `onDrop` to `^4.6.0`, and document [here](https://imzbf.github.io/md-editor-v3/en-US/docs#%F0%9F%94%96%20onDrop)

**Reply by damcuongcak3** [reply]
Thank you very much. Sorry for my silly question. Before that, I could use the drag and drop event to handle it. 🙇‍♂️

**Comment by damcuongcak3** [comment]
@imzbf I have a question. Can i upload video and preview like github markdown?

**Reply by imzbf** [reply]
This is possible, make use of `onDrop` event and `insert` api of editor.

**Reply by damcuongcak3** [reply]
Sorry. My question is about other problem, it not related with event `onDrop`.
I want to upload video to markdown and preview it. Like example

https://github.com/imzbf/md-editor-v3/assets/58088302/e7afb870-cfe3-44fa-b922-d71597694f58

**Reply by imzbf** [reply]
If it's MP4, get [extension example](https://github.com/imzbf/md-editor-extension/blob/develop/packages/v3/components/Mark/Mark.tsx) and design your extension. 

You just need to insert html like

```html
<video controls width="250">
  <source src="/media/cc0-videos/flower.webm" type="video/webm" />

  <source src="/media/cc0-videos/flower.mp4" type="video/mp4" />

  Download the
  <a href="/media/cc0-videos/flower.webm">WEBM</a>
  or
  <a href="/media/cc0-videos/flower.mp4">MP4</a>
  video.
</video>
```

I will add an extension for uploading videos later when I have time. Focus on https://github.com/imzbf/md-editor-extension
